### PR TITLE
Implemented logic for updating signer bonds swap strategy in Risk Manager

### PR DESCRIPTION
**Pull request by tomaszslabon**
This pull request adds logic for updating signer bonds swap strategy in `RiskManagerV1`.

There are two functions added:
- `beginSignerBondsSwapStrategyUpdate`
- `finalizeSignerBondsSwapStrategyUpdate`

The delay used is `GEVERNANCE_TIME_DELAY` with is 12h.

### CODE REVIEW

File: contracts/RiskManagerV1.sol [hunk 1]
```diff
@@ -226,6 +235,39 @@ contract RiskManagerV1 is Auctioneer, Ownable {
         auctionLengthChangeInitiated = 0;
     }
 
+    /// @notice Begins the signer bonds swap strategy update process.
+    /// @dev Must be followed by a finalizeSignerBondsSwapStrategyUpdate after
+    ///      the governance delay elapses.
+    /// @param _newSignerBondsSwapStrategy The new signer bonds swap strategy.
+    function beginSignerBondsSwapStrategyUpdate(
+        ISignerBondsSwapStrategy _newSignerBondsSwapStrategy
```
Comment (lukasz-zimnoch) [inline contracts/RiskManagerV1.sol:243]: Can we assert `_newSignerBondsSwapStrategy` is a non-zero address?

File: test/RiskManagerV1.test.js [hunk 2]
```diff
@@ -35,6 +35,10 @@ describe("RiskManagerV1", () => {
     signerBondsSwapStrategy = await SignerBondsSwapStrategy.deploy()
     await signerBondsSwapStrategy.deployed()
 
+    // TODO: Another type of strategy could be used
```
Comment (lukasz-zimnoch) [inline test/RiskManagerV1.test.js:38]: I think it's not relevant in the context of this test. The most important thing we need here is another address.

File: test/RiskManagerV1.test.js [hunk 3]
```diff
@@ -393,6 +397,133 @@ describe("RiskManagerV1", () => {
...
+      let tx
+      beforeEach(async () => {
+        tx = await riskManagerV1
+          .connect(owner)
+          .beginSignerBondsSwapStrategyUpdate(
+            anotherSignerBondsSwapStrategy.address
+          )
+      })
+
+      it("should not update the signerBondsSwapStrategy", async () => {
```
Comment (lukasz-zimnoch) [inline test/RiskManagerV1.test.js:411]: Can we make it more human-friendly: `should not update the signer bonds swap strategy`?

**Review by lukasz-zimnoch — COMMENTED** [review]
Looks good! Only a few tiny comments left.

File: test/RiskManagerV1.test.js (hunk 3 above)
Comment (tomaszslabon) [reply test/RiskManagerV1.test.js]: Fixed in d295b8363bac9bc886cb50c33525c461ac2c3721.

File: test/RiskManagerV1.test.js (hunk 2 above)
Comment (tomaszslabon) [reply test/RiskManagerV1.test.js]: Fixed in d295b8363bac9bc886cb50c33525c461ac2c3721.

File: contracts/RiskManagerV1.sol (hunk 1 above)
Comment (tomaszslabon) [reply contracts/RiskManagerV1.sol:243]: Fixed in d295b8363bac9bc886cb50c33525c461ac2c3721.

**Review by lukasz-zimnoch — APPROVED** [review]

**Pull request merged**